lib/pipeline/configure_test.go

```diff
@@ -48,17 +55,128 @@ var _ = Describe("Configure Pipeline", func() {
 		}
 		logger = logr.Logger{}
 
-		pipelineResources = pipeline.NewPipelineArgs("test-promise", "test-resource-request", "configure-step", "test-name", "test-namespace")
+		pipelineResources = pipeline.NewPipelineArgs("test-promise", "", "configure-step", "test-name", "test-namespace")
 	})
 
-	Describe("Pipeline Request Hash", func() {
+	Describe("Promise Configure Pipeline", func() {
 		const expectedHash = "9bb58f26192e4ba00f01e2e7b136bbd8"
+		BeforeEach(func() {
+			job, err = pipeline.ConfigurePipeline(rr, expectedHash, p, pipelineResources, "test-promise", false, logger)
+			Expect(err).NotTo(HaveOccurred())
+
+			labelsMatcher = MatchAllKeys(Keys{
+				"kratix.io/hash":                  Equal(expectedHash),
+				"kratix-workflow-action":          Equal("configure"),
+				"kratix-workflow-pipeline-name":   Equal("configure-step"),
+				"kratix.io/pipeline-name":         Equal("configure-step"),
+				"kratix-workflow-type":            Equal("promise"),
+				"kratix-workflow-kind":            Equal("pipeline.platform.kratix.io"),
+				"kratix-workflow-promise-version": Equal("v1alpha1"),
+				"kratix.io/work-type":             Equal("promise"),
+				"kratix.io/promise-name":          Equal("test-promise"),
+			})
+		})
+
+		It("creates a job with the expected metadata", func() {
+			Expect(job.ObjectMeta).To(MatchFields(IgnoreExtras, Fields{
+				"Name":      HavePrefix("kratix-test-promise-configure-step-"),
+				"Namespace": Equal("test-namespace"),
+				"Labels":    labelsMatcher,
+			}))
+		})
+
+		Context("when the pipeline name would exceed the 63 character limit", func() {
+			BeforeEach(func() {
+				promise_identifier := "long-long-long-long-promise"
+				pipeline_name := "also-very-verbose-pipeline"
+				pipelineResources = pipeline.NewPipelineArgs(promise_identifier, "", pipeline_name, "test-name", "test-namespace")
+
+				job, err = pipeline.ConfigurePipeline(rr, expectedHash, p, pipelineResources, "test-promise", false, logger)
+
+				labelsMatcher = MatchAllKeys(Keys{
+					"kratix.io/hash":                  Equal(expectedHash),
+					"kratix-workflow-action":          Equal("configure"),
+					"kratix-workflow-pipeline-name":   Equal(pipeline_name),
+					"kratix.io/pipeline-name":         Equal(pipeline_name),
+					"kratix-workflow-type":            Equal("promise"),
+					"kratix-workflow-kind":            Equal("pipeline.platform.kratix.io"),
+					"kratix-workflow-promise-version": Equal("v1alpha1"),
+					"kratix.io/work-type":             Equal("promise"),
+					"kratix.io/promise-name":          Equal(promise_identifier),
+				})
+			})
+
+			It("concatenates the pipeline name to ensure it fits the 63 character limit", func() {
```

Comment:
is the correct word for these test cases `truncate` instead of `concatenate`? 🤔

Reply:
It most certainly is